plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSourceProvider.java

```diff
@@ -720,16 +726,16 @@ public static Optional<ReaderColumns> projectBaseColumns(List<HiveColumnHandle>
 
         ImmutableList.Builder<ColumnHandle> projectedColumns = ImmutableList.builder();
         ImmutableList.Builder<Integer> outputColumnMapping = ImmutableList.builder();
-        Map<Integer, Integer> mappedHiveColumnIndices = new HashMap<>();
+        Map<String, Integer> mappedHiveBaseColumnKeys = new HashMap<>();
         int projectedColumnCount = 0;
 
         for (HiveColumnHandle column : columns) {
-            int hiveColumnIndex = column.getBaseHiveColumnIndex();
-            Integer mapped = mappedHiveColumnIndices.get(hiveColumnIndex);
+            String baseColumnKey = useColumnNames ? column.getBaseColumnName() : String.valueOf(column.getBaseHiveColumnIndex());
```

Comment:
let `mappedHiveBaseColumnKeys` be `Map<Object, ..>` and avoid stringifying the numbers.
(not pretty either way)